library/lua/dfhack.lua

```diff
@@ -86,9 +98,12 @@ function qerror(msg, level)
     dfhack.error(msg, (level or 1) + 1, false)
 end
 
+---@param ... any
 function dfhack.with_finalize(...)
     return dfhack.call_with_finalizer(0,true,...)
 end
+
+---@param ... any
 function dfhack.with_onerror(...)
```

Comment:
I agree that these type signatures could be improved by defining the known first N args. The Lua docs define them, why aren't they defined here? https://docs.dfhack.org/en/latest/docs/dev/Lua%20API.html#locking-and-finalization